### PR TITLE
Packaging test for filesystem scripts

### LINKED ISSUE #13184: bats tests should create a script and a template on the filesystem and verify that they work
This is mostly for the rpm and deb but probably also should be done for the tar tests.

Part of #13138

### PR DESCRIPTION
**Pull request by nik9000**
Adds a tests for loading scripts from the filesystem for search templates
and for search filters.

Closes #13184

### CODE REVIEW

**Comment by nik9000** [comment]
Ping @tlrx for yet more bats tests. These ones won't merge well with #13083 so we should wait until that is in before rebasing but you can review this for the general principle I guess. Or wait.

**Comment by nik9000** [comment]
Rebased and did a little more cleanup.

**Comment by tlrx** [comment]
@nik9000 I just had a look to the changes and it looks good to me.

Please let me know once conflicts are resolved so that I'll be able to run the tests.

File: qa/vagrant/src/test/resources/packaging/scripts/40_rpm_package.bats [hunk 1]
```diff
@@ -82,6 +84,8 @@ setup() {
 # Uninstall RPM package
 ##################################
 @test "[RPM] remove package" {
+    # User installed scripts aren't removed so we'll just get them ourselves
+    rm -f /etc/elasticsearch/scripts/*
```
Comment (tlrx) [inline qa/vagrant/src/test/resources/packaging/scripts/40_rpm_package.bats]: Can we use ESSCRIPTS here?

File: qa/vagrant/src/test/resources/packaging/scripts/30_deb_package.bats [hunk 2]
```diff
@@ -128,6 +130,8 @@ setup() {
 }
 
 @test "[DEB] purge package" {
+    # User installed scripts aren't removed so we'll just get them ourselves
+    rm -f /etc/elasticsearch/scripts/*
```
Comment (tlrx) [inline qa/vagrant/src/test/resources/packaging/scripts/30_deb_package.bats]: Can we use ESSCRIPTS here?

File: qa/vagrant/src/test/resources/packaging/scripts/packaging_test_utils.bash [hunk 3]
```diff
@@ -432,7 +349,15 @@ wait_for_elasticsearch_status() {
     }
 }
 
-# Executes some very basic Elasticsearch tests
+# Install scripts used to test script filters and search templates before
+# starting Elasticsearch so we don't have to wait for elasticsearch to scan for
+# them.
```
Comment (tlrx) [inline qa/vagrant/src/test/resources/packaging/scripts/packaging_test_utils.bash]: I'd put this comment directly in the .bats file rather than here

**Comment by nik9000** [comment]
@tlrx ready for review again. First commit is a rebase and the second is updates, some from comments, some from running it over and over again.

**Comment by tlrx** [comment]
@nik9000 do the tests succeed for you? I encountered various issues, mainly for `[XX PLUGINS] start elasticsearch with all plugins installed` tests

**Comment by tlrx** [comment]
OK, tests now succeed. LGTM then.

**Comment by nik9000** [comment]
> @nik9000 do the tests succeed for you? I encountered various issues, mainly for [XX PLUGINS] start elasticsearch with all plugins installed tests

That is the `--tries 60` thing I keep adding in. I broke it when I merged something else but it wasn't _consistently_ broken. Master has it so rebasing should get it. I'll do that now.

**Comment by nik9000** [comment]
Squashed and rebased. Will merge to 2.0, 2.x, and master. Thanks for the review @tlrx!

**Pull request merged**